path_sequence_planner/test/utest.cpp

```diff
@@ -48,102 +49,102 @@ TEST(IntersectTest, TestCase1)
   data2 = vtk_viewer::cutMesh(data, points2, false);
 
   // Set input mesh
-  tool_path_planner::RasterToolPathPlanner planner;
-  planner.setInputMesh(data2);
+  tool_path_planner::SurfaceWalkRasterGenerator planner;
+  planner.setInput(data2);
 
   // Set input tool data
-  tool_path_planner::ProcessTool tool;
-  tool.pt_spacing = 0.6;
-  tool.line_spacing = 0.75;
+  tool_path_planner::SurfaceWalkRasterGenerator::Config tool;
+  tool.point_spacing = 0.6;
+  tool.raster_spacing = 0.75;
   tool.tool_offset = 0.0; // currently unused
-  tool.intersecting_plane_height = 0.2; // 0.5 works best, not sure if this should be included in the tool
+  tool.intersection_plane_height = 0.2; // 0.5 works best, not sure if this should be included in the tool
   tool.min_hole_size = 0.1;
-  planner.setTool(tool);
+  planner.setConfiguration(tool);
 
   vtk_viewer::VTKViewer viz;
   std::vector<float> color(3);
 
   double scale = 0.5;
   // Display mesh results
-  color[0] = 0.9;
-  color[1] = 0.9;
-  color[2] = 0.9;
+  color[0] = 0.9f;
+  color[1] = 0.9f;
+  color[2] = 0.9f;
   viz.addPolyDataDisplay(data2, color);
 
 
   // Display surface normals
   if(DISPLAY_NORMALS)
   {
-    color[0] = 0.9;
-    color[1] = 0.1;
-    color[2] = 0.1;
+    color[0] = 0.9f;
+    color[1] = 0.1f;
+    color[2] = 0.1f;
     vtkSmartPointer<vtkPolyData> normals_data = vtkSmartPointer<vtkPolyData>::New();
-    normals_data = planner.getInputMesh();
+    normals_data = planner.getInput();
     viz.addPolyNormalsDisplay(normals_data, color, scale);
   }
 
-  tool_path_planner::ProcessPath path;
-  planner.computePaths();
-  std::vector<tool_path_planner::ProcessPath> paths = planner.getPaths();
-
+  boost::optional<tool_path_planner::ToolPaths> paths = planner.generate();
```

Comment:
in another thread I recall we were trying to move away from boost::optional. I don't recall why or what we were moving toward.

Reply:
Lets create an issue to discuss and but I would prefer to switch to return the actual type and throw an exception if something is invalid.